### PR TITLE
feat(VPicker): `flat` and `elevation` props for pickers

### LINKED ISSUE #6006: [Feature Request] Time picker, "flat mode"
### Problem to solve
Many components are available in "flat" mode to not display the shadow, but not this one.

### Proposed solution
The solution is to propose a property "flat" for this component which makes it possible to hide the shadow.

### PR DESCRIPTION
**Pull request by jacekkarczmarczyk**
image

## Description
Adds `flat` and `elevation` props to date/time/color/v-picker

## Motivation and Context
fixed #6006

## How Has This Been Tested?
playground, unit

## Markup:

<details>

```vue
<template>
  <v-container>
    <v-row>
      <v-col>
        flat
      </v-col>
      <v-col>
        default
      </v-col>
      <v-col>
        elevation 15
      </v-col>
    </v-row>
    <v-row>
      <v-col>
        <v-time-picker flat></v-time-picker>
      </v-col>
      <v-col>
        <v-time-picker></v-time-picker>
      </v-col>
      <v-col>
        <v-time-picker elevation="15"></v-time-picker>
      </v-col>
    </v-row>
    <v-row>
      <v-col>
        <v-date-picker flat></v-date-picker>
      </v-col>
      <v-col>
        <v-date-picker></v-date-picker>
      </v-col>
      <v-col>
        <v-date-picker elevation="15"></v-date-picker>
      </v-col>
    </v-row>
    <v-row>
      <v-col>
        <v-color-picker flat></v-color-picker>
      </v-col>
      <v-col>
        <v-color-picker></v-color-picker>
      </v-col>
      <v-col>
        <v-color-picker elevation="15"></v-color-picker>
      </v-col>
    </v-row>
    <v-row>
      <v-col>
        <v-picker flat><template #title>title</template>foo</v-picker>
      </v-col>
      <v-col>
        <v-picker><template #title>title</template>foo</v-picker>
      </v-col>
      <v-col>
        <v-picker elevation="15"><template #title>title</template>foo</v-picker>
      </v-col>
    </v-row>
  </v-container>
</template>
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/docs/src/lang/en/components/DatePickers.json [hunk 1]
```diff
@@ -8,6 +8,10 @@
       "desc": "Starting with year picker by default, resticting dates range and closing the picker menu after selecting the day make the perfect birthday picker.",
       "heading": "### Date pickers - birthday picker"
     },
+    "elevation": {
+      "heading": "### Picker's elevation",
+      "desc": "You can set picker's elevation using `elevation` prop or `flat` (the latter sets the elevation to 0)."
```
Comment (johnleider) [inline packages/docs/src/lang/en/components/DatePickers.json:13]:
```suggestion
      "desc": "The `v-date-picker` component supports elevation up to a maximum value of 24. For more information on elevations, visit the official [Material Design elevations](https://material.io/design/environment/elevation.html) page."
```

File: packages/docs/src/lang/en/components/TimePickers.json [hunk 2]
```diff
@@ -29,6 +29,10 @@
       "heading": "### AM/PM switch in title",
       "desc": "You can move AM/PM switch to picker's title."
     },
+    "elevation": {
+      "heading": "### Picker's elevation",
+      "desc": "You can set picker's elevation using `elevation` prop or `flat` (the latter sets the elevation to 0)."
```
Comment (johnleider) [inline packages/docs/src/lang/en/components/TimePickers.json:34]:
```suggestion
      "desc": "Emphasize the `v-time-picker` component by providing an **elevation** from 0 to 24. Elevation modifies the `box-shadow` css property."
```

Comment (jacekkarczmarczyk) [reply packages/docs/src/lang/en/components/TimePickers.json:34]: is different text than for `v-date-picker` intended?

Comment (johnleider) [reply packages/docs/src/lang/en/components/TimePickers.json:34]: yes

**Comment by jacekkarczmarczyk** [comment]
> In addition, I'm not convinced that pickers should have the flat prop if their default elevation is 0. It wouldn't overwrite the user provided elevation. If they did <v-picker elevation="24" flat /> the picker className would resemble class="elevation-0 elevation-24".

@johnleider Picker's behaviour is same as for `v-card`, default elevation is 1 (although it's not defined in props but in default styles), also it adds `*--flat` prop for flat pickers (not `elevation-0`).

```vue
<template>
  <v-container>
    <v-card class="my-2"><v-card-text>default</v-card-text></v-card>
    <v-card class="my-2" flat><v-card-text>flat</v-card-text></v-card>
    <v-card class="my-2" elevation="10"><v-card-text>elevation 10</v-card-text></v-card>
    <v-card class="my-2" elevation="10" flat><v-card-text>flat + elevation 10</v-card-text></v-card>

    <v-picker full-width class="my-2"><v-card-text>default</v-card-text></v-picker>
    <v-picker full-width class="my-2" flat><v-card-text>flat</v-card-text></v-picker>
    <v-picker full-width class="my-2" elevation="10"><v-card-text>elevation 10</v-card-text></v-picker>
    <v-picker full-width class="my-2" elevation="10" flat><v-card-text>flat + elevation 10</v-card-text></v-picker>
  </v-container>
</template>
```
image

Defining both `elevation` (!= 0) and `flat` props for a component sounds like misuse, but `flat` prop could fail indeed if we have defaults props mechanism - if you specify default `elevation` to some value != 0 then `flat` prop won't do anything (unless you manually unset value of `elevation` prop: `<foo :elevation="undefined" flat>`, that sounds like a separate issue with elevatable mixin, not related to this PR imho

**Pull request merged**